php-practice/section12.php

```diff
@@ -159,3 +214,54 @@ class TwelveSidedDie extends Dice {
  * HINT: What shared methods or properties need to be in the parent HatOfNames class?
  * HINT: What visibility do these shared methods and properties need? (public, private, or protected?)
  */
+
+abstract class HatOfNames {
+  const MODE_RETAIN_WINNER = 1;
+  const MODE_NEW_WINNER_EVERY_TIME = 2;
+
+  protected $names = [];
+  protected $winner;
+  protected $mode;
```

Comment:
Yeah I agree. I'm thinking tomorrow sometime. I'll let you know. Andi and I have to do some grocery shopping. We did a little bit today but put of the rest until tomorrow :)